src/wasm/extract.cpp

```diff
@@ -656,29 +657,37 @@ std::string extractor::extract(const std::string& list_json) {
   }
 
   switch (extraction_settings_.language_filter_options) {
-    case LanguageFilterOptions::All: {
-      log_info << "Language filter option: All";
-      break;
-    }
-    case LanguageFilterOptions::LanguageAgnostic: {
-      log_info << "Language filter option: Language agnostic";
-      break;
-    }
-    case LanguageFilterOptions::SelectedLanguage: {
-      log_info << "Language filter option: Selected language";
-      break;
-    }
-    case LanguageFilterOptions::SelectedLanguageAndAgnostic: {
-      log_info << "Language filter option: Selected language and agnostic";
-      break;
-    }
-    default: {
-      log_info << "Language filter option: Unknown";
-      break;
-    }
+  case LanguageFilterOptions::All: {
+    log_info << "Language filter option: All";
+    break;
+  }
+  case LanguageFilterOptions::LanguageAgnostic: {
+    log_info << "Language filter option: Language agnostic";
+    break;
+  }
+  case LanguageFilterOptions::SelectedLanguage: {
+    log_info << "Language filter option: Selected language";
+    break;
+  }
+  case LanguageFilterOptions::SelectedLanguageAndAgnostic: {
+    log_info << "Language filter option: Selected language and agnostic";
+    break;
+  }
+  default: {
+    log_info << "Language filter option: Unknown";
+    break;
+  }
   }
 
   auto resolved_files = resolve_collisions(selected_files, lang);
+  if (aborted) {
+    log_info << "Extraction aborted";
+    abort_zip();
+
+    json ret;
+    ret["status"] = "Aborted by user";
```

Comment:
The messages should probably read something like "Aborted: error due to file conflict" to differentiate from regular user-induced abort.